graf2d/graf/src/TGaxis.cxx

```diff
@@ -1974,7 +1974,7 @@ void TGaxis::PaintAxis(Double_t xmin, Double_t ymin, Double_t xmax, Double_t yma
                      if (fNModLabs) ChangeLabelAttributes(k+1, nlabels, textaxis, chtemp);
                      typolabel = chtemp;
                      if (!optionTime) typolabel.ReplaceAll("-", "#minus");
-                     if (autotoff) {
+                     if (autotoff && (gPad->GetWw()*gPad->GetWNDC() != 0.0)) {
```

Comment:
```suggestion
                     double scale = gPad->GetWw()*gPad->GetWNDC();
                     if (autotoff && (scale != 0.0)) {
```
and then re-use `scale` in the call to `TMath::Max`.

Reply:
Also, I would have assumed that the proper change would have been the call to `TMath::Max`:
```
if (scale > 0.)
   toffset = TMath::Max(toffset,(double)w/scale);
```
Why do you need to protect the whole branch, i.e. why is `scale == 0.0` equivalent to `!autotoff`?

Reply:
Done.

>the whole branch
That is an "optimization".  My understanding was that the actions in the branch were doene only to calculate the `toffset`. So I've assumed `scale=0` eliminates the need to do that. But your suggestion is better.

Best regards,

Andrii